### PR TITLE
A few improvements for Tabs

### PR DESCRIPTION
**Pull request by devongovett**
A few things I noticed when testing out tabs:

1. Multiple tabs with the same item keys would produce conflicting ids on the page. Added a base id for the tabs instance so this is avoided.
2. Ensure the selected tabs receives focus when tabbing into the tablist, rather than only the first item. This was actually an issue in the selection hooks, so would apply across all collections.
3. Ensure a tab is always selected, even when no `defaultSelectedKey` is passed. This will select the first tab by default.

### CODE REVIEW

File: packages/@react-spectrum/tabs/stories/Tabs.stories.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ storiesOf('Tabs', module)
 
 function render(props = {}) {
   return (
-    <Tabs defaultSelectedKey="val1" {...props} onSelectionChange={action('onSelectionChange')}>
+    <Tabs {...props} maxWidth={500} onSelectionChange={action('onSelectionChange')}>
```
Comment (snowystinger) [inline packages/@react-spectrum/tabs/stories/Tabs.stories.js:68]: did you want to apply this to the icon stories as well? currently they stretch the entire page
Also, should we use a spectrum value?

**Comment by devongovett** [comment]
Haven't touched css yet. This is mostly just a demo of aria so far.

File: packages/@react-spectrum/tabs/stories/Tabs.stories.js (hunk 1 above)
Comment (devongovett) [reply packages/@react-spectrum/tabs/stories/Tabs.stories.js:68]: Added.

**Comment by horacioh** [comment]
awesome work!!

is there a reason why there is no docs page for `useTabs`? I checked also the generated documentation on this PR and there is no Tabs docs inside the react-aria package.

thanks in advance!

**Comment by snowystinger** [comment]
@horacioh no docs yet because this will be an alpha and we don't have a feature flag system in our docs yet, so these will get docs when they get their first semver release

**Comment by horacioh** [comment]
> @horacioh no docs yet because this will be an alpha and we don't have a feature flag system in our docs yet, so these will get docs when they get their first semver release

awesome!! thanks for the info @snowystinger

**Pull request merged**